src/libcrun/container.c

```diff
@@ -3502,6 +3502,70 @@ libcrun_container_update_from_file (libcrun_context_t *context, const char *id,
   return libcrun_container_update (context, id, content, len, err);
 }
 
+static int
+compare_update_values (const void *a, const void *b)
+{
+  const struct libcrun_update_value_s *aa = a;
+  const struct libcrun_update_value_s *bb = b;
+  int ret;
+
+  ret = strcmp (aa->section, bb->section);
+  if (ret)
+    return ret;
+  return strcmp (aa->name, bb->name);
+}
+
+int
+libcrun_container_update_from_values (libcrun_context_t *context, const char *id,
+                                      struct libcrun_update_value_s *values, size_t len,
+                                      libcrun_error_t *err)
+{
+  const char *current_section = "";
+  const unsigned char *buf;
+  yajl_gen gen = NULL;
+  size_t i, buf_len;
+  int ret;
+
+  gen = yajl_gen_alloc (NULL);
+  if (gen == NULL)
+    return crun_make_error (err, errno, "yajl_gen_create failed");
+  yajl_gen_map_open (gen);
+
+  qsort (values, len, sizeof(struct libcrun_update_value_s), compare_update_values);
+
+  for (i = 0; i < len; i++)
+    {
+      if (strcmp (values[i].section, current_section))
```

Comment:
Personally I would initialize current_section to NULL and add a null check here, but maybe something else guarantees that value[].section is neverm ""?